converter/grpc_interceptor.go

```diff
@@ -71,3 +74,49 @@ func NewPayloadCodecGRPCClientInterceptor(options PayloadCodecGRPCClientIntercep
 		},
 	})
 }
+
+type NewFailureGRPCClientInterceptorOptions struct {
+	// DataConverter ??
+	EncodeCommonAttributes bool
+}
+
+// Do we want to expose this from converter so that we are sure things line up?
+type encodedFailure struct {
+	Message    string `json:"message"`
+	StackTrace string `json:"stack_trace"`
+}
+
+// NewFailureGRPCClientInterceptor returns a GRPC Client Interceptor that will mimic the encoding
+// that the SDK system would perform when configured with a FailureConverter with the EncodeCommonAttributes option set.
+func NewFailureGRPCClientInterceptor(options NewFailureGRPCClientInterceptorOptions) (grpc.UnaryClientInterceptor, error) {
+	if !options.EncodeCommonAttributes {
+		return nil, fmt.Errorf("EncodeCommonAttributes must be set for this interceptor to function")
+	}
+
+	dc := GetDefaultDataConverter()
+	return proxy.NewFailureVisitorInterceptor(proxy.FailureVisitorInterceptorOptions{
+		Outbound: &proxy.VisitFailuresOptions{
+			Visitor: func(vpc *proxy.VisitFailuresContext, failure *failurepb.Failure) error {
+				var err error
+				failure.EncodedAttributes, err = dc.ToPayload(encodedFailure{
+					Message:    failure.Message,
+					StackTrace: failure.StackTrace,
+				})
+				return err
+			},
+		},
+		Inbound: &proxy.VisitFailuresOptions{
+			Visitor: func(vpc *proxy.VisitFailuresContext, failure *failurepb.Failure) error {
+				var ea encodedFailure
+				err := dc.FromPayload(failure.EncodedAttributes, &ea)
```

Comment:
Should not assume this will always be set. Some older workflow histories may come through this proxy without having this set. Just check if present and try to decode.

Take this logic from `internal/failure_decoder.go` which just ignores if it can't decode the attributes. (or even better, make a small helper you both share)